core/ordersync/ordersync.go

```diff
@@ -340,6 +346,39 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 	return nil
 }
 
+// PeriodicallyGetOrders periodically calls GetOrders. It waits a minimum of
+// approxDelay (with some random jitter) between each call. It will block until
+// there is a critical error or the given context is canceled.
+func (s *Service) PeriodicallyGetOrders(ctx context.Context, minPeers int, approxDelay time.Duration) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		if err := s.GetOrders(ctx, minPeers); err != nil {
+			return err
+		}
+
+		// Note(albrow): The random jitter here helps smooth out the frequency of ordersync
+		// requests and helps prevent a situation where a large number of nodes are requesting
+		// orders at the same time.
+		delay := calculateDelayWithJitter(approxDelay, ordersyncJitterAmount)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(delay):
+		}
+	}
+}
+
+func calculateDelayWithJitter(approxDelay time.Duration, jitterAmount float64) time.Duration {
+	jitterBounds := int(float64(approxDelay) * jitterAmount * 2)
+	delta := rand.Intn(jitterBounds) - jitterBounds/2
```

Comment:
From https://golang.org/pkg/math/rand/#NormFloat64, `0 <= rand.Intn(jitterBounds) < jitterBounds`. Given this, the bounds given in the comment above `ordersyncJitterAmount` should be

```
approxDelay * (1 - jitter) <= actualDelay < approxDelay * (1 + jitter)
```
Unfortunately, `jitter + math.SmallestNonzeroFloat{32,64} == jitter` in most cases because the floating point numbers are too many orders of magnitude different, so it could be tricky to make this inclusive. I'd say that we just update the comment.

Reply:
I just made the comment more accurate by changing the the upper bound to exclusive. There's no inherent reason to prefer an inclusive upper bound.

Reply:
Thanks, yep I agree that this is just as good as an inclusive upper bound.